### PR TITLE
chore: update packages and readme

### PR DESCRIPTION
**Pull request by jkaster**
- updated README
- created README templates for each specific extension type
- updated package refs for the 4 different templates
- updated React `<HelloWorld />` component templates
- added `yarn new` for local template development
- also added optional `Nix` support
- linty fresh

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if has nix; then
+  use nix
+fi
```
Comment (jeremytchang) [inline .envrc:3]: Missing newline?

Comment (jkaster) [reply .envrc:3]: It doesn't matter for processing the env file but I'll add it if I have to push to the PR again. Still working on figuring out why CI is failing when tests are fine locally.

File: src/utils.js [hunk 2]
```diff
@@ -52,13 +52,13 @@ const generatorPrompt = (projectName) => {
...
         let parsed = value.toLowerCase();
-        if (parsed == "no framework") {
+        if (parsed === "no framework") {
           parsed = "vanilla";
         }
         return parsed;
       },
     },
-  ]).catch((e) => {
+  ]).catch((_e) => {
```
Comment (bryans99) [inline src/utils.js:61]: Below it says see other log lines for details. If we ignore the error will we see the details? Should we console log the error?

File: templates/react-typescript/README.md.template [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+   Typically, the model is named the the same as the extension project. The model is used to control access to the extension.
+
+   - [Configure the model you created](https://docs.looker.com/data-modeling/getting-started/create-projects#configuring_a_model) so that it has access to some connection (any connection).
+
+7. Connect the new project to Git.
+
+   This can be done in multiple ways:
+
+   - Create a new repository on GitHub or a similar service, and follow the instructions to [connect your project to Git](https://docs.looker.com/data-modeling/getting-started/setting-up-git-connection)
+   - A simpler but less powerful approach is to set up git with the "Bare" repository option which does not require connecting to an external Git Service.
```
Comment (bryans99) [inline templates/react-typescript/README.md.template:76]: Remove bare repo reference?

File: templates/vanilla-javascript/README.md.template [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+   Typically, the model is named the the same as the extension project. The model is used to control access to the extension.
+
+   - [Configure the model you created](https://docs.looker.com/data-modeling/getting-started/create-projects#configuring_a_model) so that it has access to some connection (any connection).
+
+7. Connect the new project to Git.
+
+   This can be done in multiple ways:
+
+   - Create a new repository on GitHub or a similar service, and follow the instructions to [connect your project to Git](https://docs.looker.com/data-modeling/getting-started/setting-up-git-connection)
+   - A simpler but less powerful approach is to set up git with the "Bare" repository option which does not require connecting to an external Git Service.
```
Comment (bryans99) [inline templates/vanilla-javascript/README.md.template:76]: Remove bare repo

File: templates/vanilla-typescript/README.md.template [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+   Typically, the model is named the the same as the extension project. The model is used to control access to the extension.
+
+   - [Configure the model you created](https://docs.looker.com/data-modeling/getting-started/create-projects#configuring_a_model) so that it has access to some connection (any connection).
+
+7. Connect the new project to Git.
+
+   This can be done in multiple ways:
+
+   - Create a new repository on GitHub or a similar service, and follow the instructions to [connect your project to Git](https://docs.looker.com/data-modeling/getting-started/setting-up-git-connection)
+   - A simpler but less powerful approach is to set up git with the "Bare" repository option which does not require connecting to an external Git Service.
```
Comment (bryans99) [inline templates/vanilla-typescript/README.md.template:76]: Remove bare repo

**Review by bryans99 — APPROVED** [review]
LGTM but perhaps a little more doc cleanup

Maybe we go overboard with the git repo stuff. Maybe just refer to Looker documentation

File: templates/react-javascript/README.md.template [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    }
+   \`\`\`
+
+6. Create a \`model\` LookML file in your project.
+
+   Typically, the model is named the the same as the extension project. The model is used to control access to the extension.
+
+   - Add a connection in this model (it can be any connection).
+   - [Configure the model you created](https://docs.looker.com/data-modeling/getting-started/create-projects#configuring_a_model) so that it has access to some connection.
```
[the remark this answers is not in the document]
Comment (jkaster) [reply templates/react-javascript/README.md.template]: Updated

**Review by bryans99 — APPROVED** [review]
LGTM but you still got at least three 'the the' s

File: src/utils.js (hunk 2 above)
Comment (jkaster) [reply src/utils.js:61]: I'll add logging of the error anyway. All I did was the `_` for the lint report

File: README.md [hunk 7]
```diff
@@ -16,7 +16,15 @@ cd <your-project-name>
 yarn develop
 ```
 
-_Developing? To run this project locally, just run `./src/index.js <your-project-name>` from the root directory_
+### Developing this template
```
[the remark this answers is not in the document]
Comment (jkaster) [reply README.md:19]: Described it now

File: templates/react-javascript/README.md.template [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+7. Connect the new project to Git.
+
+   This can be done in multiple ways:
+
+   - Create a new repository on GitHub or a similar service, and follow the instructions to [connect your project to Git](https://docs.looker.com/data-modeling/getting-started/setting-up-git-connection)
+   - A simpler but less powerful approach is to set up git with the "Bare" repository option which does not require connecting to an external Git Service.
+
+8. Commit the changes and deploy them to production through the Project UI.
+
+9. Reload the page and click the \`Browse\` dropdown menu. You should see the extension label in the list.
```
[the remark this answers is not in the document]
Comment (jkaster) [reply templates/react-javascript/README.md.template:77]: Do you want me to list the other UI method?

**Review by bryans99 — APPROVED** [review]
LGTM

**Review by bryans99 — APPROVED** [review]

**Review by bryans99 — APPROVED** [review]
LGTM

**Pull request merged**